tests/integration/test_provider.py

```diff
@@ -99,3 +130,51 @@ async def test_remove_application_removes_user(ops_test: OpsTest, usernames):
                 zookeeper_uri=zookeeper_uri,
                 model_full_name=ops_test.model_full_name,
             )
+
+
+@pytest.mark.abort_on_fail
+async def test_change_client_topic(ops_test: OpsTest):
+    await ops_test.model.units.get(f"{DUMMY_NAME_2}/0").run_action("change-topic")
+
+    time.sleep(20)
```

Comment:
You can remove the sleep, with an approach like below. This is optional, just a suggestion.
```python
    action = await ops_test.model.units.get(f"{DUMMY_NAME_2}/0").run_action("change-topic")
    await action.wait()
```